final_task/rss_parser/Output_functions.py

```diff
@@ -0,0 +1,332 @@
+    logging.info("Converting to json view!")
+    news_dict = {
+        "Source": the_feed.get('feed', '').get('title'),
+        "Number of news": num_of_news,
+        "News": [{"number": item.number_of_novelty,
+                  "title": item.title_of_novelty,
+                  "published": item.time_of_novelty,
+                  "link": item.source_link,
+                  "description": item.description,
+                  "images links": item.images_links,
+                  "alternative text": item.alt_text,
+                  "corrected time": item.date_corrected,
+                  "main source": item.main_source,
+                  } for num, item in enumerate(pack_of_news)]
+    }
+    logging.info("Converted to json view!")
+    return json.dumps(news_dict)
+
+
+def converting_novelty_to_json(item):
+    """
+    Transfer novelty from novelty view to json view to put it in the file
+    for future using when creating cache pack of news
+    :param item:
+    :return:
+    """
+    news_dict = {
+        "number": item.number_of_novelty,
+        "title": item.title_of_novelty,
+        "published": item.time_of_novelty,
+        "link": item.source_link,
+        "description": item.description,
+        "images links": item.images_links,
+        "alternative text": item.alt_text,
+        "main source": item.main_source,
+    }
+    return json.dumps(news_dict)
+
+
+def getting_info_into_file(item):
+    """
+    Preparing information to be written into the file in more readable way
+    :param item:
+    :return:
+    """
+    novelty = "\n{0}.\nTitle: {1}\nDate: {2}\nLink: {3}\nDescription:\n {4}\nImages links:{5}\nAlternative text:{6}\n" \
+              "Main source: {7}" \
+        .format(item.number_of_novelty,
+                pprint.pformat(item.title_of_novelty, width=115),
+                item.time_of_novelty,
+                pprint.pformat(item.source_link, width=115),
+                pprint.pformat(item.description, width=115),
+                pprint.pformat(item.images_links),
+                pprint.pformat(item.alt_text, width=115),
+                item.main_source)
+    return novelty
+
+
+def getting_corrected_time(item):
+    """
+    Getting time in view %Y%m%d
+    :param item:
+    :return:
+    """
+    corrected_date = datetime.strptime(item.get('published', ''), '%a, %d %b %Y %X %z')
+    return corrected_date.strftime('%Y%m%d')
+
+
+def getting_time_for_json(item):
+    corrected_date = datetime.strptime(item, '%a, %d %b %Y %X %z')
+    return corrected_date.strftime('%Y%m%d')
+
+
+def reading_file(name_of_file):
+    with open(name_of_file, 'r', encoding='utf-8') as news_cache:
+        return news_cache.read()
+
+
+def writing_to_file(pack_of_news, filename):
+    """
+    Writing information into 2 files: News_cache.txt and News_cache_json.txt
+    Creating 2 files because it's easier to read information into computer from JSON file than another file
```

Comment:
A question:

Why do you need 2 files?

Reply:
Because then I read file to get cache information into list in my program (creating list of objects of Novelty class) to find news by date and source. If it looks silly I can somehow refuse using two files and leave only one